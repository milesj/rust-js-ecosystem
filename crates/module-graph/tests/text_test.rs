mod utils;

use starbase_sandbox::{assert_snapshot, create_sandbox};
use utils::generate_graph_for_file;

mod text {
    use super::*;

    #[test]
    fn graphql() {
        let sandbox = create_sandbox("text");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "graphql.gql"));
    }

    #[test]
    fn graphql_alt_extension() {
        let sandbox = create_sandbox("text");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "graphql.graphql"));
    }

    #[test]
    fn html() {
        let sandbox = create_sandbox("text");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "index.html"));
    }

    #[test]
    fn less() {
        let sandbox = create_sandbox("text");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "styles.less"));
    }

    #[test]
    fn map() {
        let sandbox = create_sandbox("text");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "source.js.map"));
    }

    #[test]
    fn sass() {
        let sandbox = create_sandbox("text");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "styles.sass"));
    }

    #[test]
    fn scss() {
        let sandbox = create_sandbox("text");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "styles.scss"));
    }

    #[test]
    fn stylus() {
        let sandbox = create_sandbox("text");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "styles.styl"));
    }

    #[test]
    fn svg() {
        let sandbox = create_sandbox("text");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "icon.svg"));
    }
}
