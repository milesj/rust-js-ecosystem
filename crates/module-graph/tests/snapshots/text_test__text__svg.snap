---
source: crates/module-graph/tests/text_test.rs
expression: "generate_graph_for_file(sandbox.path(), \"icon.svg\")"
---
{
    1: Module {
        exports: [],
        fragment: None,
        id: 1,
        imports: [],
        package_name: None,
        path: "/root/icon.svg",
        query: None,
        source: TextModule {
            kind: Svg,
            source: "<svg xmlns=\"http://www.w3.org/2000/svg\" xml:space=\"preserve\" viewBox=\"0 0 84 84\">\n  <path d=\"m68.73333,28.58l0,10.54c0,1.55 -0.83,2.98 -2.17,3.76l-11.43,6.6c-1.34,0.77 -3,0.77 -4.34,0l-11.43,-6.6a4.342,4.342 0 0 1 -2.17,-3.76l0,-13.19c0,-1.55 0.83,-2.98 2.17,-3.76l11.43,-6.6c1.34,-0.77 3,-0.77 4.34,0l3.69,2.13a36.534,36.534 0 0 0 -21.62,-7.06c-20.25,0 -36.67,16.42 -36.67,36.67s16.42,36.67 36.67,36.67s36.67,-16.42 36.67,-36.67c0.01,-6.84 -1.88,-13.24 -5.14,-18.73z\"/>\n  <path d=\"m83.13333,24.31l0,-13.19c0,-1.55 -0.83,-2.98 -2.17,-3.76l-11.43,-6.6c-1.34,-0.77 -3,-0.77 -4.34,0l-11.43,6.6a4.342,4.342 0 0 0 -2.17,3.76l0,13.19c0,1.55 0.83,2.98 2.17,3.76l11.43,6.6c1.34,0.77 3,0.77 4.34,0l11.43,-6.6a4.334,4.334 0 0 0 2.17,-3.76z\"/>\n</svg>\n",
        },
    },
}
